test/multipart-fileLimit.test.js

```diff
@@ -316,3 +316,127 @@ test('should NOT throw fileSize limitation error when throwFileSizeLimit is glob
     t.assert.ifError(error)
   }
 })
+
+test('should throw fileSize limitation error when used alongside attachFieldsToBody and set request config', async function (t) {
+  t.plan(1)
+
+  const fastify = Fastify()
+  t.after(() => fastify.close())
+
+  fastify.register(multipart, {
+    attachFieldsToBody: true
+  })
+
+  const randomFileBuffer = Buffer.alloc(2_000_000)
+  crypto.randomFillSync(randomFileBuffer)
+
+  fastify.post('/', {
+    config: {
+      multipartOptions: {
+        limits: {
+          fileSize: 1_000_000
```

Comment:
@grigoran 

The file size limitation error isn’t caused by this configuration.
In this test, it should occur only when the request configuration is set, as described in the title.

Reply:
@grigoran 
The config should trigger the error as described in the message:
> should throw fileSize limitation error when used alongside attachFieldsToBody and set request config

Reply:
@grigoran 

The test should be updated so that the error is caused by the config, as the message states:
> should throw fileSize limitation error when used alongside attachFieldsToBody and set request config